src/canvasObject.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var CanvasObject = function(config) {
+  this.fullyOpened = config.fullyOpened || false;
+  this.visible = config.visible || true;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
+
+  this.id = config.canvas['@id'];
+  this.height = config.canvas.height;
+  this.width = config.canvas.width;
+  this.images = config.canvas.images;
+  this.label = config.canvas.label;
+  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
+};
+
+CanvasObject.prototype = {
+  openTileSource: function(viewer) {
+    // We've already loaded this tilesource instead of the thumbnail
+    if(this.fullyOpened) {
+      return;
+    }
+    var self = this;
+    var bounds = self.mainImageObj.getBounds();
```

Comment:
I don't know that I have good reasons to back this up, but my preference is to not use `self` in places where you don't need it, e.g. this line. My preference would be for this line to use `this` and the code inside the success function to use `self` (where it's necessary). 

Also, I guess just a stylistic thing, but I like having the `var self = this` line always at the top of the function if it's present...that way you don't have to go hunting for it. 

More of my `self` thoughts: http://blog.iangilman.com/2015/02/self-this.html